internal/install/packs/packs_installer.go

```diff
@@ -0,0 +1,137 @@
+package packs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+
+	"github.com/newrelic/newrelic-cli/internal/credentials"
+	"github.com/newrelic/newrelic-cli/internal/install/types"
+	"github.com/newrelic/newrelic-cli/internal/install/ux"
+	"github.com/newrelic/newrelic-client-go/newrelic"
+	"github.com/newrelic/newrelic-client-go/pkg/dashboards"
+	"github.com/newrelic/newrelic-client-go/pkg/entities"
+	log "github.com/sirupsen/logrus"
+)
+
+// Maybe both fetchPacks and installPacks should be one interface
+// vs separate interfaces?
+// nolint: golint
+type PacksInstaller interface {
+	Install(ctx context.Context, packs []types.OpenInstallationObservabilityPack) error
+}
+
+type ConcretePacksInstaller struct {
+	client            *newrelic.NewRelic
+	progressIndicator ux.ProgressIndicator
+}
+
+func NewConcretePacksInstaller(client *newrelic.NewRelic) *ConcretePacksInstaller {
+	return &ConcretePacksInstaller{
+		client:            client,
+		progressIndicator: ux.NewPlainProgress(),
+	}
+}
+
+func defaultHTTPGetFunc(dashboardURL string) (*http.Response, error) {
+	return http.Get(dashboardURL)
+}
+
+func (p *ConcretePacksInstaller) Install(ctx context.Context, packs []types.OpenInstallationObservabilityPack) error {
+	msg := "Installing observability packs"
+	p.progressIndicator.Start(msg)
+	defer p.progressIndicator.Stop()
+
+	for _, pack := range packs {
+		fmt.Printf("\n  Installing pack: %s\n", pack.Name)
+
+		// Only installing dashboards currently
+		if pack.Dashboards != nil {
+			for _, dashboard := range pack.Dashboards {
+				if _, err := p.createObservabilityPackDashboard(ctx, dashboard); err != nil {
+					p.progressIndicator.Fail(msg)
+					return fmt.Errorf("Failed to create observability pack dashboard [%s]: %s", dashboard.Name, err) // nolint: golint
+				}
+			}
+		}
+	}
+
+	p.progressIndicator.Success(msg)
+	return nil
+}
+
+func (p *ConcretePacksInstaller) createObservabilityPackDashboard(ctx context.Context, d types.OpenInstallationObservabilityPackDashboard) (*dashboards.DashboardCreateResult, error) {
+	defaultProfile := credentials.DefaultProfile()
+	accountID := defaultProfile.AccountID
+
+	body, err := getJSONfromURL(d.URL)
```

Comment:
This should ideally be implemented in the client, with types and potentially API methods generated by tutone.  Create an issue for the work?

Reply:
Agreed that we should do this eventually. From last meeting, sounds like we're going to punt on this for now.

Reply:
Issue created: https://github.com/newrelic/newrelic-client-go/issues/748